### PR TITLE
v_3_0_0: update nginx-ingress-controller container to 0.10.1

### PR DESCRIPTION
**Pull request by corest**
Towards: https://github.com/giantswarm/giantswarm/issues/2374

### CODE REVIEW

**Comment by puja108** [comment]
We need to be careful about breaking changes between our beta and this version. I remember they included a change from the old annotation to a new one, which can be reverted by setting some flags.

We need to decide if we want to tell users to move all their annotations (if they have them) to the new one, or we set the IC to have old annotation style. @othylmann (and customer team, e.g. @pipo02mix) WDYT?

**Comment by corest** [comment]
@puja108 I went through the changelog and don't see any changes except prometheus annotations. Did you mean those?

**Comment by puja108** [comment]
their on the 0.9 final release: https://github.com/kubernetes/ingress-nginx/releases/tag/nginx-0.9.0

**Comment by puja108** [comment]
I talked to @marians and we think that we should go with upstream default and include the new annotation in our changelog and announcement for v3. This way if a customer in the future looks at upstream nginx docs they won't be confused because of different annotation prefix.

/cc @teemow @othylmann

**Comment by pipo02mix** [comment]
Then in case we want to upgrade an existing cluster (IC component to latest version too) we need to encourage the customer to change all annotations, right?

And it would provoke downtime IIRC

**Comment by marians** [comment]
It should be possible to avoid downtime (mostly) by first adding the annotations in the new form before the upgrade, then do the IC upgrade, then remove the old annotations.

File: v_3_0_0/master_template.go [hunk 1]
```diff
@@ -644,14 +644,25 @@ write_files:
                         - nginx-ingress-controller
                   topologyKey: kubernetes.io/hostname
           serviceAccountName: nginx-ingress-controller
+          initContainers:
+          - command:
+            - sh
+            - -c
+            - sysctl -w net.core.somaxconn=32768; sysctl -w net.ipv4.ip_local_port_range="1024 65535"
```
Comment (corest) [inline v_3_0_0/master_template.go]: @r7vme I remember you did some manual changes for ingress-controller. Were those somehow related to the queue for connections waiting for acceptance?

**Comment by corest** [comment]
tested last coredns/nginx ingress controller. Works fine, didn't find any issues

**Comment by corest** [comment]
Towards https://github.com/giantswarm/giantswarm/issues/2374

**Pull request merged**